teps/0008-api-spec.md

```diff
@@ -0,0 +1,115 @@
+
+#### Client Author
+
+As with the end user above, a client author (e.g., a CLI or UI tool author, or platform author building on top of Tekton) can opt to depend on Tekton API features when building their integration with the understanding of which features they're depending on might restrict implementation portability.
+
+#### Platform Implementor
+
+A new platform implementor can implement a conformant Tekton API implementation since the requirements are clearly documented and can be tested using the test suite. They can choose to implement recommended surfaces if they choose to, but can  assert conformance-compliance for required features, with test results to back it up.
+
+### Risks and Mitigations
+
+The API spec might prove to be so overly-limited that users and clients can't meaningfully depend on a useful and conformant Tekton implementation, which would likely lead them to depend on the feature set of the existing Kubernetes implementation, making this the de facto spec.
+
+Since we are defining the spec while we only have one real implementation, based on Kubernetes, we might accidentally require features in the spec that are easy to satisfy in our one implementation, but which might be problematic for future implementations. This could limit the success of future implementations and potentially harm the ecosystem. We can mitigate this by carefully and very conservatively considering all additions to required/recommended surfaces in the API.
+
+A second or third API implementation may never materialize, and the work to document and enforce conformance may be wasted. By limiting initial requirements we can both lower the barrier to entry for future implementations, and limit the amount of work we expend on the conformance test suite, until such a time as the benefit of future implementations makes further investment necessary.
+
+## Test Plan
+
+We should add a separate test suite to the Tekton Pipelines repo that sends requests to a specified Tekton Pipelines implementation and expects that the request is accepted and that expected behavior is observed as a result.
+
+For example, sending a request to create a `TaskRun` to the API that only specifies required fields should succeed against any conformant implementation. Periodically polling that `TaskRun` resource until it indicates it completed successfully (or otherwise) should also succeed against any conformant implementation.
+
+To be clear, this test suite doesn't need to focus as much on _correctness_ -- that's what existing end-to-end tests cover -- but should only send simple requests and expect to observe conformant behavior.
```

Comment:
am i right to assume we'd elaborate on this in a future conformance tep?

Reply:
I think we'd elaborate on the specifics of this in the PR(s) that add the conformance tests.